assets/src/blocks/TopicLink/TopicLinkEditor.js

```diff
@@ -0,0 +1,255 @@
+   * @return {Object} An object with x and y properties as decimal values.
+   */
+  const setFocalPoint = focalPoints => {
+    const [x, y] = focalPoints.split(' ').map(value => parseFloat(value) / 100);
+    return {x, y};
+  };
+
+  const selectedCategory = setBlockCategory();
+
+  // Update attributes with image data for frontend
+  setAttributes({
+    imageUrl: imageUrl || '',
+    imageAlt: imageAlt || '',
+  });
+
+  /**
+   * Renders the block preview in the editor.
+   * @return {JSX.Element} The block preview element.
+   */
+  const renderEditInPlace = () => (
+    <section className="topic-link-block">
+      <div className="background-image">
+        {imageUrl && (
+          <img
+            src={imageUrl}
+            alt={imageAlt}
+            style={{objectPosition: focal_points}}
+          />
+        )}
+      </div>
+      <div className="topic-link-content">
+        <p>Learn more about {selectedCategory?.name || ''}</p>
+        <div className="chevron-icon"></div>
+      </div>
+    </section>
+  );
+
+  /**
+   * Renders the Inspector Controls for the block.
+   * @return {JSX.Element} The Inspector Controls component.
+   */
+  const addInspectorControls = () => (
+    <InspectorControls>
+      <PanelBody title={__('Settings', 'planet4-blocks-backend')}>
+        <SelectControl
+          label={__('Select Category:', 'planet4-blocks-backend')}
+          value={categoryId}
+          options={[...categoriesList.map(category => ({label: category.name, value: category.id}))]}
+          onChange={id => setAttributes({categoryId: parseInt(id)})}
+        />
+        <MediaUploadCheck>
+          <MediaUpload
+            title={__('Select Background Image', 'planet4-blocks-backend')}
+            type="image"
+            onSelect={({id}) => setAttributes({imageId: id})}
+            value={imageId}
+            allowedTypes={['image']}
+            render={({open}) => (
+              <Button onClick={open} className="button">
+                {imageId ? __('Change Background Image', 'planet4-blocks-backend') : __('Select Background Image', 'planet4-blocks-backend')}
+              </Button>
+            )}
+          />
+        </MediaUploadCheck>
+        {imageUrl && (
+          <div className="wp-block-master-theme-gallery__FocalPointPicker">
+            <strong className="components-base-control__help">
+              {__('Select image focal point', 'planet4-blocks-backend')}
+            </strong>
+            <p className="components-base-control__help">
+              {__('Adjust the “Left” and “Top” fields to position the focal point of the image in percentage, where 0% is the top/left edge and 100% is the bottom/right edge.', 'planet4-blocks-backend')}
+            </p>
+            <FocalPointPicker
+              url={imageUrl}
+              dimensions={{width: 400, height: 100}}
+              value={setFocalPoint(focal_points)}
+              onChange={value => setObjectPosition(value)}
```

Comment:
I think that this can be shortened to `onChange={setObjectPosition}`

Reply:
Done!